src/middleware/companyAuth.middleware.ts

```diff
@@ -0,0 +1,75 @@
+import ApplicationLogger from 'ch-logging/lib/ApplicationLogger'
+import { Session } from 'ch-node-session-handler'
+import { SessionKey } from 'ch-node-session-handler/lib/session/keys/SessionKey'
+import { SignInInfoKeys } from 'ch-node-session-handler/lib/session/keys/SignInInfoKeys'
+import { ISignInInfo } from 'ch-node-session-handler/lib/session/model/SessionInterfaces'
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+
+import { DISSOLUTION_SESSION_KEY } from 'app/constants/app.const'
+import DissolutionSession from 'app/models/dissolutionSession'
+import { Mutable } from 'app/models/mutable'
+import Optional from 'app/models/optional'
+import { JwtEncryptionService } from 'app/services/encryption/jwtEncryption.service'
+
+export interface AuthConfig {
+  accountUrl: string,
+  accountRequestKey: string,
+  accountClientId: string,
+  chsUrl: string,
+}
+
+const OATH_SCOPE_PREFIX = 'https://api.companieshouse.gov.uk/company/'
+
+export default function CompanyAuthMiddleware(authConfig: AuthConfig,
+                                              encryptionService: JwtEncryptionService,
+                                              logger: ApplicationLogger): RequestHandler {
+  return async (req: Request, res: Response, next: NextFunction) => {
+      const dissolutionSession: Optional<DissolutionSession> = req.session!.getExtraData(DISSOLUTION_SESSION_KEY)
+      if (!dissolutionSession?.companyNumber) {
+        return next(new Error('No Company Number in session'))
+      }
+      const companyNumber: Optional<string> = dissolutionSession!.companyNumber
+      const signInInfo: ISignInInfo = req.session!.get<ISignInInfo>(SessionKey.SignInInfo) || {}
```

Comment:
Do you need the default `|| {}`? Assuming the AuthMiddleware will ensure that this is definitely present